From 0a0b0e32141c91c33847e185828b275fbb10dbeb Mon Sep 17 00:00:00 2001
From: Matthijs van Duin <matthijsvanduin@gmail.com>
Date: Wed, 13 Feb 2019 09:07:16 +0100
Subject: [PATCH 4/6] ARM: dts: dra7/am335x: add outline definitions for pruss

These can be replaced by the proper ones once remoteproc-pru has been
ported to 4.19.
---
 arch/arm/boot/dts/am33xx.dtsi | 19 +++++++++++++
 arch/arm/boot/dts/dra7.dtsi   | 50 +++++++++++++++++++++++++++++++++++
 2 files changed, 69 insertions(+)

diff --git a/arch/arm/boot/dts/am33xx.dtsi b/arch/arm/boot/dts/am33xx.dtsi
index 4690ac4fbe4f..2f32e9b1b3e9 100644
--- a/arch/arm/boot/dts/am33xx.dtsi
+++ b/arch/arm/boot/dts/am33xx.dtsi
@@ -719,6 +719,25 @@
 			ti,timer-pwm;
 		};
 
+		pruss_soc_bus: pruss_soc_bus@4a326004 {
+			reg = <0x4a326004 0x4>;
+			ti,hwmods = "pruss";
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges = <0x0 0x4a300000 0x80000>;
+			status = "disabled";
+
+			pruss: pruss@0 {
+				reg = <0 0x80000>;
+				interrupt-parent = <&intc>;
+				interrupts = <20 21 22 23 24 25 26 27>;
+				interrupt-names = "host2", "host3", "host4",
+				                  "host5", "host6", "host7",
+				                  "host8", "host9";
+				status = "disabled";
+			};
+		};
+
 		rtc: rtc@44e3e000 {
 			compatible = "ti,am3352-rtc", "ti,da830-rtc";
 			reg = <0x44e3e000 0x1000>;
diff --git a/arch/arm/boot/dts/dra7.dtsi b/arch/arm/boot/dts/dra7.dtsi
index 8a4d1d1077ae..a9aedaaa578e 100644
--- a/arch/arm/boot/dts/dra7.dtsi
+++ b/arch/arm/boot/dts/dra7.dtsi
@@ -1184,6 +1184,56 @@
 			status = "disabled";
 		};
 
+		pruss_soc_bus1: pruss_soc_bus@4b226004 {
+			reg = <0x4b226004 0x4>;
+			ti,hwmods = "pruss1";
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges = <0x0 0x4b200000 0x80000>;
+			status = "disabled";
+
+			pruss1: pruss@0 {
+				reg = <0 0x80000>;
+				interrupts = <GIC_SPI 186 IRQ_TYPE_LEVEL_HIGH>,
+				             <GIC_SPI 187 IRQ_TYPE_LEVEL_HIGH>,
+				             <GIC_SPI 188 IRQ_TYPE_LEVEL_HIGH>,
+				             <GIC_SPI 189 IRQ_TYPE_LEVEL_HIGH>,
+				             <GIC_SPI 190 IRQ_TYPE_LEVEL_HIGH>,
+				             <GIC_SPI 191 IRQ_TYPE_LEVEL_HIGH>,
+				             <GIC_SPI 192 IRQ_TYPE_LEVEL_HIGH>,
+				             <GIC_SPI 193 IRQ_TYPE_LEVEL_HIGH>;
+				interrupt-names = "host2", "host3", "host4",
+				                  "host5", "host6", "host7",
+				                  "host8", "host9";
+				status = "disabled";
+			};
+		};
+
+		pruss_soc_bus2: pruss_soc_bus@4b2a6004 {
+			reg = <0x4b2a6004 0x4>;
+			ti,hwmods = "pruss2";
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges = <0x0 0x4b280000 0x80000>;
+			status = "disabled";
+
+			pruss2: pruss@0 {
+				reg = <0 0x80000>;
+				interrupts = <GIC_SPI 196 IRQ_TYPE_LEVEL_HIGH>,
+				             <GIC_SPI 197 IRQ_TYPE_LEVEL_HIGH>,
+				             <GIC_SPI 198 IRQ_TYPE_LEVEL_HIGH>,
+				             <GIC_SPI 199 IRQ_TYPE_LEVEL_HIGH>,
+				             <GIC_SPI 200 IRQ_TYPE_LEVEL_HIGH>,
+				             <GIC_SPI 201 IRQ_TYPE_LEVEL_HIGH>,
+				             <GIC_SPI 202 IRQ_TYPE_LEVEL_HIGH>,
+				             <GIC_SPI 203 IRQ_TYPE_LEVEL_HIGH>;
+				interrupt-names = "host2", "host3", "host4",
+				                  "host5", "host6", "host7",
+				                  "host8", "host9";
+				status = "disabled";
+			};
+		};
+
 		abb_mpu: regulator-abb-mpu {
 			compatible = "ti,abb-v3";
 			regulator-name = "abb_mpu";
-- 
2.20.1

